### DISCUSSION TITLE
iperf3 3.9 problem with ack (ubuntu, link via vdsl)

### DISCUSSION THREAD

**Discussion by wchpikus** [esnet/iperf#1292] [Q&A]
hello
i have strange behaviour with iperf3.
client ubuntu 18.04, server also 18,04
All test with other link (lan, gpon) works ok, but no with vdls+ on some device.
I made some wireshark debug, and i see lost ack form server to client.
Second ack and psk lost...why?
On server side after disconnect client i see error (iperf size of data read does not correspond to offered length).
attached wireshark 

Client:
![lost-client](https://user-images.githubusercontent.com/3776183/158059686-8f65e12d-1c4e-4dd8-99b8-eaf02789e041.png)

Server:
![server](https://user-images.githubusercontent.com/3776183/158059688-596d4d97-57d2-4090-832f-ef75870644d9.png)

any suggestion, what is wrong?

**Comment by davidBar-On** [comment]
Hi, I am not familiar with vdsl+, but it seems that the problem is related to this interface, the implementation of TCP over this interface or something like this.  As you wrote, messages are lost between the client and the server, even short ones.

It would help a lot if you can get Wireshark logs also from the server side, so it will be possible to compare the sent and received messages on each side.  That sometimes can give hints to the problem.  Also, the Wireshark logs themselves (e.g. pcap files) can help to better analyze what happens.

**Comment by wchpikus** [comment]
hi, tnx for replay.
Tomorrow i will take some data from server and clients. no problem.
I do not have on the same setup any problems with LAN and Gpon but vdsl+ on some device looks as upper.
Just iperf stop.
If i allow to connect to external source (internet or something) it works.
UDP also work, but no with iperf.
I see dates is incoming to lan, pass via dsl, pass router input-output, pass to interface on server iperf, but iperf do not connect to client (do not receiving data...
I thing some communication client-server using tcpip lost, on higher port (60602??)  
Maybe it take some time, pass data via vdsl+ and iperf need quick answer, when setup connection client-server and client trying resend  PSH to fast...i do not know. 
First graph is from client, second from server

**Comment by wchpikus** [comment]
this is log from Wireshark.
client 10.100.35.8
server 10.100.35.10
Btw, ping working to bought site:)

[vdsl2-wireshark.zip](https://github.com/esnet/iperf/files/8242470/vdsl2-wireshark.zip)

**Comment by davidBar-On** [comment]
Hi @[wchpikus](https://github.com/wchpikus), there seem to be a problem with the network you use, that is not related to iperf3.  Although the first two SYN/ACK message is o.k., the messages send by the server in lines 15 and 16 does not arrive to the client.  Only when the client's re-transmission in line 5 arrives to the server in line 19, the ACK sent in line 20 arrives to the client in line 6.  From that point, not of the server's messages arrive to the client for about 8 seconds and the client terminates the connection (line 10).

Note the (successful) ARP request from the server.  I assume it is because there was no ack to the TCP keep-alive messages from the server, so the server assumed that something may be wrong with its routing tables.

**Comment by wchpikus** [comment]
hi
I know that, for some season this ack was lost...
what is strange, iper2 working (i have some scores, but it is not what I expect), but communication client-server is ok.
i test connection using nc on ubuntu, tcp and udp working well...strange.
Udp using iperf2 also working..

**Comment by davidBar-On** [comment]
Hi, what I notice is that the client did not receive any of the server's messages with length less than 60.  That includes the acks in line 17,18 and their re-transmissions in lines 21-25.  That may happen if such messages are not padded when sent over Ethernet to be 60 bytes (Ethernet minimum message size is 64 bytes - 60 + 4 bytes checksum).

Note that Client to Server path does not have that problem - the 54 bytes message sent in the Client line 3 arrives to the server as 60 bytes message in line 15 (with 8 padding bytes).

To further evaluate the issue, it will help if you can send additional information:
1. What is "vdsl+"?  I am not familiar with it and I was not able to find an explanation.
2. Is there a VLAN from the server to the client?  I once saw a case where the Ethernet padding was added to the message with the 4 bytes VLAN tag, so when the tag was removed the length was only 56 and the message was discarded.
3. Can you send Wireshark logs for a successful iperf2 session (at least log from the server side)?  It will help to see whether iperf2 also sends messages less than 60 bytes. (If it does and they arrive to the client than my guess about the root of the problem is incorrect.)
